packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -0,0 +1,255 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import {
+	createElement,
+	Fragment,
+	useCallback,
+	useEffect,
+	useReducer,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { closeSmall, external } from '@wordpress/icons';
+import { useWooBlockProps } from '@woocommerce/block-templates';
+import {
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlMenu as Menu,
+	__experimentalSelectControlMenuItem as MenuItem,
+	useAsyncFilter,
+	Spinner,
+} from '@woocommerce/components';
+import { Product } from '@woocommerce/data';
+import { getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { FormattedPrice } from '../../../components/formatted-price';
+import { ProductEditorBlockEditProps } from '../../../types';
+import {
+	getLoadLinkedProductsDispatcher,
+	getRemoveLinkedProductDispatcher,
+	getSearchProductsDispatcher,
+	getSelectSearchedProductDispatcher,
+	reducer,
+} from './reducer';
+import { LinkedProductListBlockAttributes } from './types';
+
+export function getProductImageStyle( product: Product ) {
+	return product.images.length > 0
+		? {
+				backgroundImage: `url(${ product.images[ 0 ].src })`,
+		  }
+		: undefined;
+}
+
+export function Edit( {
+	attributes,
+	context: { postType },
+}: ProductEditorBlockEditProps< LinkedProductListBlockAttributes > ) {
+	const { property } = attributes;
+	const blockProps = useWooBlockProps( attributes );
+	const [ state, dispatch ] = useReducer( reducer, {
+		linkedProducts: [],
+		searchedProducts: [],
+	} );
+	const loadLinkedProductsDispatcher =
+		getLoadLinkedProductsDispatcher( dispatch );
+	const searchProductsDispatcher = getSearchProductsDispatcher( dispatch );
+	const selectSearchedProductDispatcher =
+		getSelectSearchedProductDispatcher( dispatch );
+	const removeLinkedProductDispatcher =
+		getRemoveLinkedProductDispatcher( dispatch );
+
+	const [ linkedProductIds, setLinkedProductIds ] = useEntityProp< number[] >(
```

Comment:
Is it possible to use `useProductEntityProp` here so that we could support using a meta data value?

I'm not sure how the prop type is handled? (How does it convert the prop to a `number[]` in this case?

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/43145/commits/4af1c1abe8d346a728ba78299f4e868f8f5fac0a